File: python/maximum-xor-for-each-query.py
class Solution:
    def getMaximumXor(self, nums: List[int], maximumBit: int) -> List[int]:
        answer = []
        current = (2 ** maximumBit) - 1
        for num in nums:
            current ^= num
        for i in range(len(nums)):
            answer.append(current)
            current ^= nums[-1 - i]
        return answer

File: go/maximum-xor-for-each-query.go
import (
    "math"
)

func getMaximumXor(nums []int, maximumBit int) []int {
    answer := make([]int, 0)
    current := int(math.Pow(2, float64(maximumBit))) - 1;
    for i := 0; i < len(nums); i++ {
        current ^= nums[i]
    }
    for i := 0; i < len(nums); i++ {
        answer = append(answer, current)
        current ^= nums[len(nums) - i - 1]
    }
    return answer
}
